File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/RelatedLettersDB/RelatedLetters/RelatedLettersColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace CorrespondenceSystem.RelatedLettersDB.Columns;

[ColumnsScript("RelatedLettersDB.RelatedLetters")]
[BasedOnRow(typeof(RelatedLettersRow), CheckNames = true)]
public class RelatedLettersColumns
{
    [EditLink, DisplayName("Letter Identifire"), AlignRight]
    
    public string LetterIdentifier { get; set; }
   
    public short RelationType { get; set; }
    public DateTime CreatedDate { get; set; }
    

}

File: CorrespondenceSystem/CorrespondenceSystem.Web/Modules/ServerTypes/RelatedLettersDB/RelatedLettersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { RelatedLettersRow } from "./RelatedLettersRow";

export interface RelatedLettersColumns {
    LetterIdentifier: Column<RelatedLettersRow>;
    RelatedLetterTitle: Column<RelatedLettersRow>;
    RelationType: Column<RelatedLettersRow>;
    CreatedDate: Column<RelatedLettersRow>;
}

export class RelatedLettersColumns extends ColumnsBase<RelatedLettersRow> {
    static readonly columnsKey = 'RelatedLettersDB.RelatedLetters';
    static readonly Fields = fieldsProxy<RelatedLettersColumns>();
}